### PR TITLE
blkid: add helpers that get gpt partition uuid as sd_id128_t

### PR DESCRIPTION
**Pull request by poettering**
just some refactoring to make things simpler.

### CODE REVIEW

File: src/partition/repart.c [hunk 1]
```diff
@@ -4889,16 +4889,13 @@ static int resolve_copy_blocks_auto_candidate(
                 return false;
         }
 
-        t = blkid_partition_get_uuid(pp);
-        if (isempty(t)) {
-                log_debug("Partition %u:%u has no UUID.",
-                          major(partition_devno), minor(partition_devno));
+        r = blkid_partition_get_uuid_id128(pp, &u);
+        if (r == -ENXIO) {
+                log_debug_errno(r, "Partition %u:%u has no UUID.", major(partition_devno), minor(partition_devno));
```
Comment (yuwata) [inline src/partition/repart.c]: `DEVNUM_FORMAT_STR` can be used.

File: src/partition/repart.c [hunk 2]
```diff
@@ -4889,16 +4889,13 @@ static int resolve_copy_blocks_auto_candidate(
...
                 return false;
         }
-
-        r = sd_id128_from_string(t, &u);
         if (r < 0) {
-                log_debug_errno(r, "Failed to parse partition UUID \"%s\": %m", t);
+                log_debug_errno(r, "Failed to read partition UUID off %u:%u: %m", major(partition_devno), minor(partition_devno));
```
Comment (yuwata) [inline src/partition/repart.c]: - `"off"` -> `"of"`
- `DEVNUM_FORMAT_STR` can be used.

File: src/shared/blkid-util.h [hunk 3]
```diff
@@ -4,7 +4,29 @@
 #if HAVE_BLKID
 #  include <blkid.h>
 
+#  include "sd-id128.h"
 #  include "macro.h"
```
Comment (yuwata) [inline src/shared/blkid-util.h]: Please insert an empty line after the `sd` header.

File: src/shared/blkid-util.h [hunk 4]
```diff
@@ -4,7 +4,29 @@
...
+#  include "string-util.h"
 
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(blkid_probe, blkid_free_probe, NULL);
+
+static inline int blkid_partition_get_uuid_id128(blkid_partition p, sd_id128_t *ret) {
```
Comment (yuwata) [inline src/shared/blkid-util.h:14]: missing assertion for `p`

File: src/shared/blkid-util.h [hunk 5]
```diff
@@ -4,7 +4,29 @@
...
+        const char *s;
+
+        s = blkid_partition_get_uuid(p);
+        if (isempty(s))
+                return -ENXIO;
+
+        return sd_id128_from_string(s, ret);
+}
+
+static inline int blkid_partition_get_type_id128(blkid_partition p, sd_id128_t *ret) {
```
Comment (yuwata) [inline src/shared/blkid-util.h:26]: Also here.

File: src/shared/dissect-image.c [hunk 6]
```diff
@@ -627,22 +627,22 @@ static int dissect_image(
...
                                 continue;
+                        }
 
-                        stype = blkid_partition_get_type_string(pp);
-                        if (!stype)
-                                continue;
-                        if (sd_id128_from_string(stype, &type_id) < 0)
+                        r = blkid_partition_get_type_id128(pp, &type_id);
+                        if (r < 0) {
+                                log_debug_errno(r, "Failed to read partitiont type UUID, ignoring: %m");
```
Comment (yuwata) [inline src/shared/dissect-image.c]: `"partitiont"` -> `"partition"`

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by poettering** [comment]
updated, ptal

**Review by DaanDeMeyer — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

**Pull request merged**